code/game/objects/effects/decals/remains.dm

```diff
@@ -27,12 +27,26 @@
 	desc = "They look like the remains of a small reptile."
 	icon_state = "lizard"
 
-/obj/effect/decal/remains/attack_hand(mob/user as mob)
-	to_chat(user, "<span class='notice'>[src] sinks together into a pile of ash.</span>")
-	var/turf/simulated/floor/F = get_turf(src)
+//Target turns to ash.
+/proc/crumble(user, source)
+	to_chat(user, "<span class='notice'>[source] sinks together into a pile of ash.</span>")
+	var/turf/simulated/floor/F = get_turf(source)
 	if (istype(F))
 		new /obj/effect/decal/cleanable/ash(F)
-	qdel(src)
+	qdel(source)
+
+/obj/effect/decal/remains/Move()
+	if(src.pulledby)
+		crumble(pulledby, src)
+	else
+		return
```

Comment:
Few things for tidiness:
You can remove `src` in front of pulledby. Code automatically assumes you're referencing the `src` unless otherwise specified (Sort of like how you didn't have to do src.crumble())!
If you follow my other suggestions, you can just turn `crumble(pulledby, src)` into `crumble()`
No need to have that else-return statement. If there isn't any pulledby, nothing special happens.